plan/config.go

```diff
@@ -104,10 +114,54 @@ func parseBehavior(d string) Behavior {
 		ClientAxis: clientAxis,
 		ParamsAxes: values,
 	}
-
 	return behavior
 }
 
+func validateAndApplyFilters(filterMap map[string][]Filter, behaviors Behaviors) (Behaviors, error) {
+	for i := range behaviors {
+		behavior := &behaviors[i]
+		filters := filterMap[behavior.Name]
+		for _, filter := range filters {
+			for match := range filter.AxisMatches {
+				if match != behavior.ClientAxis && !axisContainsRule(match, behavior.ParamsAxes) {
```

Comment:
`if !behavior.HasAxis(axis) { err }` 

I believe you can then replace axisContainsRule by this new HasAxis method.